### PR TITLE
Login view

### PR DESCRIPTION
**Pull request by johangv**

### CODE REVIEW

File: lib/features/login/domain/repositories/login_repository.dart [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import 'dart:ffi';
+
+import 'package:firebase_auth/firebase_auth.dart';
+
+abstract class LoginRepository {
+  Future<UserCredential> signIn();
+  Future<void> logOut();
+}
```
Comment (oscarshaitan) [inline lib/features/login/domain/repositories/login_repository.dart]: no need of abstract classs if there is no plan to more than 1 class to implement.
/iknow that CLEAN said to do it but also clean could easily be a bazuca for a mosquito

File: lib/features/login/presentation/cubit/login_cubit.dart [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+class LoginCubit extends Cubit<LoginState> {
+  LoginCubit(this._loginRepository) : super(const Initial());
+
+  final LoginRepository _loginRepository;
+
+  void login() async {
+    final result = await _loginRepository.signIn();
+    if (result.user != null) {
+      emit(const Success());
```
Comment (oscarshaitan) [inline lib/features/login/presentation/cubit/login_cubit.dart]: is more common the notation LoginState.success()

File: lib/features/splash/data/local/splash_repository_imp.dart [hunk 3]
```diff
@@ -1,9 +1,13 @@
+import 'package:firebase_auth/firebase_auth.dart';
 import 'package:fiura_ecosystem/features/splash/domain/repositories/splash_repository.dart';
 
 class SplashRepositoryImp extends SplashRepository {
+  final FirebaseAuth auth = FirebaseAuth.instance;
```
Comment (oscarshaitan) [inline lib/features/splash/data/local/splash_repository_imp.dart:5]: this should be provided on the constructor by the get it

File: lib/features/login/data/local/login_repository_imp.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+class LoginRepositoryImp extends LoginRepository {
+  //Collections
+  final String users = "users";
+
+  //Cloud firestore
+  final FirebaseFirestore db = FirebaseFirestore.instance;
+
+  //Firabase Auth
+  final GoogleSignIn googleSignIn = GoogleSignIn();
+  final FirebaseAuth auth = FirebaseAuth.instance;
```
Comment (oscarshaitan) [inline lib/features/login/data/local/login_repository_imp.dart]: this should be provided on the constructor by the get it

**Review by oscarshaitan — CHANGES_REQUESTED** [review]
if you can add a video or photos of the feature should be awesome

File: lib/features/login/presentation/cubit/login_cubit.dart (hunk 2 above)
Comment (johangv) [reply lib/features/login/presentation/cubit/login_cubit.dart]: Done

File: lib/features/splash/data/local/splash_repository_imp.dart (hunk 3 above)
Comment (johangv) [reply lib/features/splash/data/local/splash_repository_imp.dart:5]: Fixed

File: lib/features/login/data/local/login_repository_imp.dart (hunk 4 above)
Comment (johangv) [reply lib/features/login/data/local/login_repository_imp.dart]: Fixed

File: lib/features/login/domain/repositories/login_repository.dart (hunk 1 above)
Comment (johangv) [reply lib/features/login/domain/repositories/login_repository.dart]: Fixed

**Comment by johangv** [comment]
This is the video of the feature: [https://drive.google.com/file/d/1k_L_1XGJyhRXFL8vkVLYC5JovptE_v4z/view?usp=share_link](url)

**Pull request merged**